src/livecodes/core.ts

```diff
@@ -1318,29 +1319,40 @@ const share = async (
         },
       }
     : config;
-  const contentParam = shortUrl
-    ? '?x=id/' +
-      (await shareService.shareProject({
-        ...content,
-        result: includeResult ? getCache().result : undefined,
-      }))
-    : '?x=code/' + compress(JSON.stringify(content));
 
   const currentUrl = (location.origin + location.pathname).split('/').slice(0, -1).join('/') + '/';
-
   const url = permanentUrl ? permanentUrlService.getAppUrl() : currentUrl;
+  const shareURL = new URL(url);
+  if (shortUrl) {
+    shareURL.search =
+      'x=id/' +
+      (await shareService.shareProject({
+        ...content,
+        result: includeResult ? getCache().result : undefined,
+      }));
+  } else {
+    const hashParams = compress(JSON.stringify(content));
+    shareURL.hash = 'x=code/' + hashParams;
 
-  const shareURL = url + contentParam;
+    const searchParams = new URLSearchParams();
+    if (content.title && content.title != defaultConfig.title) {
```

Comment:
please use `!==` instead of `!=` to [make eslint happy](https://github.com/live-codes/livecodes/actions/runs/14988103780/job/42110921403?pr=819) :)